src/java/org/apache/cassandra/index/sai/disk/v1/PerColumnIndexFiles.java

```diff
@@ -50,6 +50,11 @@ public PerColumnIndexFiles(IndexDescriptor indexDescriptor, IndexContext indexCo
             files.put(IndexComponent.POSTING_LISTS, indexDescriptor.createPerIndexFileHandle(IndexComponent.POSTING_LISTS, indexContext));
             files.put(IndexComponent.TERMS_DATA, indexDescriptor.createPerIndexFileHandle(IndexComponent.TERMS_DATA, indexContext));
         }
+        else
+        {
+            files.put(IndexComponent.KD_TREE, indexDescriptor.createPerIndexFileHandle(IndexComponent.KD_TREE, indexContext));
```

Comment:
Nit: The condition of this if block could be `indexContext.isLiteral()` instead of `TypeUtil.isLiteral(indexContext.getValidator())`.